teos/src/gatekeeper.rs

```diff
@@ -567,65 +540,98 @@ mod tests {
         let available_slots = gatekeeper
             .add_update_appointment(user_id, uuid, &appointment)
             .unwrap();
+        // Simulate the watcher adding the appointment in the database.
+        gatekeeper
+            .dbm
+            .lock()
+            .unwrap()
+            .store_appointment(uuid, &appointment)
+            .unwrap();
 
-        assert!(gatekeeper.registered_users.lock().unwrap()[&user_id]
-            .appointments
-            .contains_key(&uuid));
+        let (_, user_locators) = gatekeeper.get_user_info(user_id).unwrap();
+        assert!(user_locators.contains(&appointment.locator()));
         assert_eq!(slots_before, available_slots + 1);
 
-        // Slots should have been updated in the database too. Notice the appointment won't be there yet
-        // given the Watcher is responsible for adding it, and it will do so after calling this method
+        // Slots should have been updated in the database too.
         let mut loaded_user = gatekeeper.dbm.lock().unwrap().load_user(user_id).unwrap();
         assert_eq!(loaded_user.available_slots, available_slots);
 
-        // Adding the exact same appointment should leave the slots count unchanged
+        // Adding the exact same appointment should leave the slots count unchanged.
+        // We don't really need to update the appointment in the DB since it's the very same appointment.
         let mut updated_slot_count = gatekeeper
             .add_update_appointment(user_id, uuid, &appointment)
             .unwrap();
-        assert!(gatekeeper.registered_users.lock().unwrap()[&user_id]
-            .appointments
-            .contains_key(&uuid));
+
+        let (_, user_locators) = gatekeeper.get_user_info(user_id).unwrap();
+        assert!(user_locators.contains(&appointment.locator()));
         assert_eq!(updated_slot_count, available_slots);
+
         loaded_user = gatekeeper.dbm.lock().unwrap().load_user(user_id).unwrap();
         assert_eq!(loaded_user.available_slots, updated_slot_count);
 
         // If we add an update to an existing appointment with a bigger data blob (modulo ENCRYPTED_BLOB_MAX_SIZE), additional slots should be taken
+        #[allow(clippy::redundant_clone)]
         let mut bigger_appointment = appointment.clone();
```

Comment:
Clippy is complaining about this line that we should remove the `.clone`: `note: cloned value is neither consumed nor mutated`.
But the cloned value is indeed mutated. Am I missing something?

Reply:
nvm, only my outdated clippy had this false positive.